### PR TITLE
Update handling of account authentication expired error

### PR DESCRIPTION
**Pull request by wallyqs**
Applies similar handling `-ERR 'Account Authentication Expired` error as with others auth expired errors.

Signed-off-by: Waldemar Quevedo <wally@synadia.com>

### CODE REVIEW

**Comment by wallyqs** [comment]
After this merge, this test should be updated in the server: https://github.com/nats-io/nats-server/compare/master...wallyqs:auth-acc-expired-err

File: nats_test.go [hunk 1]
```diff
@@ -1651,6 +1651,107 @@ func TestRevokedUserCredentials(t *testing.T) {
 	wg.Wait()
 }
 
+func TestExpiredAccountAuthentication(t *testing.T) {
```
Comment (wallyqs) [inline nats_test.go:1654]: Think some of these tests should be refactored into table driven tests since too similar, I'll update the PR

**Comment by kozlovic** [comment]
@wallyqs I don't understand.. you already merged a PR that did that: https://github.com/nats-io/nats.go/pull/684

**Comment by wallyqs** [comment]
@kozlovic there were three types of expired/revoked errors that had to be handled in the same way, I changed into table driven tests so that it is easier to add/remove these special cases /cc @matthiashanel 
```go
		{"expired users credentials", AUTHENTICATION_EXPIRED_ERR, ErrAuthExpired},
		{"revoked users credentials", AUTHENTICATION_REVOKED_ERR, ErrAuthRevoked},
		{"expired account", ACCOUNT_AUTHENTICATION_EXPIRED_ERR, ErrAccountAuthExpired},
```

**Review by kozlovic — APPROVED** [review]
LGTM (sorry I got confused with 684 and thought that you added the same auth error that was there, but it is different).

**Review by matthiashanel — APPROVED** [review]
LGTM

**Pull request merged**